### PR TITLE
Add Custom Response Header Logic and the Unit Tests for it

### PR DESCRIPTION
**Pull request by songrx1997**

### CODE REVIEW

File: pkg/backends/features/customresponseheaders.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (swetharepakula) [inline pkg/backends/features/customresponseheaders.go:2]: nit: change to 2022

**Comment by swetharepakula** [comment]
Thanks for the change Ruixian!

A couple of small nits, otherwise lgtm! Please rebase on master and update the PR.

@spencerhance, please review as well. Thanks!

**Comment by songrx1997** [comment]
> Thanks for the change Ruixian!
> 
> A couple of small nits, otherwise lgtm! Please rebase on master and update the PR.
> 
> @spencerhance, please review as well. Thanks!

Thanks @swetharepakula!  Just updated the PR for nits.

**Review by spencerhance — APPROVED** [review]
Thanks @songrx1997!

Generally lgtm.  We will have to be somewhat cautious in rolling this out since I think it will wipe out any existing response headers that somewhat has set manually.  (Unsupported behavior but still a risk factor we've seen before).

**Pull request merged**